Sources/NIOSSH/Connection State Machine/Operations/AcceptsUserAuthMessages.swift

```diff
@@ -41,7 +41,7 @@ extension AcceptsUserAuthMessages {
         let result = try self.userAuthStateMachine.receiveUserAuthRequest(message)
 
         if let future = result {
-            return .possibleFutureMessage(future.map(Self.transform(_:)))
+            return .possibleFutureMessage(future.map{ Self.transform($0, username: message.username) } )
```

Comment:
We probably need to amend the user auth state machine to keep track of the username for which the user has been authenticating. We can do this in a circular buffer (to maintain ordering). When we complete the user auth we know what the user authenticated for and can store this on the relevant state.

As with the connection state machine, we should also expose a computed property that lets the connection state machine ask what the username was. It can then pass that through to other states.

Reply:
I'm trying a few things here, but I don't think ordering will work. The many outstanding public key requests to the authentication delegate can complete in any order. I've got cleaner implementation going, but fundamentally it still involves pairing the ultimate future with a username.

Reply:
The user must complete them in order though, as the wire protocol has no way to associate each request with a response.